addons/irlight/functions/fnc_initItemContextMenu.sqf

```diff
@@ -25,23 +25,26 @@
             private _baseClass = getText (configFile >> "CfgWeapons" >> _item >> "baseWeapon");
             _item != _baseClass + _variant
         }, {
-            params ["", "", "_item", "", "_variant"];
+            params ["_unit", "", "_item", "_slot", "_variant"];
 
             private _baseClass = getText (configFile >> "CfgWeapons" >> _item >> "baseWeapon");
 
-            ACE_player removePrimaryWeaponItem _item;
-            ACE_player addPrimaryWeaponItem (_baseClass + _variant);
-            playSound "click";
-
-            if (_turnedOn) then {
-                // Force update of flashlight
-                ACE_player action ["GunLightOff", ACE_player];
-
-                {
-                    ACE_player action ["GunLightOn", ACE_player];
-                    ACE_player action ["IRLaserOn", ACE_player];
-                } call CBA_fnc_execNextFrame;
+            switch (_slot) do {
+                case "RIFLE_POINTER": {
+                    _unit removePrimaryWeaponItem _item;
+                    _unit addPrimaryWeaponItem (_baseClass + _variant);
```

Comment:
looking at cba PR: https://github.com/CBATeam/CBA_A3/pull/1154
> Currently switching from one flashlight to another in the same frame whilst the flashlight is on causes the light effect not to update to the new flashlight

I'm not sure if this is still an issue, 
but I have seen occasionally weird behavior with the IR flashlights showing the halo when not in NV
not sure if related to this

![20231213195231_1](https://github.com/user-attachments/assets/89583869-6637-47c4-8f70-36f8cd8d6654)

Reply:
> I'm not sure if this is still an issue [...]

I'm pretty certain it's still an issue. It's why I turn off the the previous weapon's laser and light before turning on the new one in https://github.com/acemod/ACE3/pull/10118/files#diff-59b5c2cffa7beb08924bf8f3cf64796c5974b0c8114181355f155575894a8eeaR38

Reply:
Now requires #10135 to solve that problem.